src/modules/event-listener/implementation/ot-ethers/ot-ethers.js

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable no-await-in-loop */
+import ethers from 'ethers';
+import OtEventListener from '../ot-event-listener.js';
+
+import {
+    MAXIMUM_NUMBERS_OF_BLOCKS_TO_FETCH,
+    BLOCK_TIME_MILLIS,
+    WS_RPC_PROVIDER_PRIORITY,
+    HTTP_RPC_PROVIDER_PRIORITY,
+    FALLBACK_PROVIDER_QUORUM,
+    RPC_PROVIDER_STALL_TIMEOUT,
+    MAX_BLOCKCHAIN_EVENT_SYNC_OF_HISTORICAL_BLOCKS_IN_MILLS,
+} from '../../../../constants/constants.js';
+
+class OtEthers extends OtEventListener {
+    async initialize(config, logger) {
+        await super.initialize(config, logger);
+    }
+
+    async initializeBlockchainEventListener(blockchainConfig) {
+        this.blockchainConfig = blockchainConfig;
+        await this.initializeRpcProvider();
+    }
+
+    async initializeRpcProvider() {
+        const providers = [];
+        for (const rpcEndpoint of this.blockchainConfig.rpcEndpoints) {
+            const isWebSocket = rpcEndpoint.startsWith('ws');
+            const Provider = isWebSocket
+                ? ethers.providers.WebSocketProvider
+                : ethers.providers.JsonRpcProvider;
+            const priority = isWebSocket ? WS_RPC_PROVIDER_PRIORITY : HTTP_RPC_PROVIDER_PRIORITY;
+
+            try {
+                const provider = new Provider(rpcEndpoint);
+                // eslint-disable-next-line no-await-in-loop
+                await provider.getNetwork();
+
+                let isArchiveNode = false;
+                try {
+                    const block = await provider.getBlock(1);
+
+                    if (block) {
+                        isArchiveNode = true;
+                    }
+                } catch (error) {
+                    /* empty */
+                }
+
+                if (isArchiveNode) {
+                    providers.push({
+                        provider,
+                        priority,
+                        weight: 1,
+                        stallTimeout: RPC_PROVIDER_STALL_TIMEOUT,
+                    });
+                } else {
+                    this.logger.warn(`${rpcEndpoint} RPC is not an Archive Node, skipping...`);
```

Comment:
should you add return here? so it doesn't log "Connected to the blockchain RPC" if it's skipped

Reply:
not necesarrily return but move connected to the if above

Reply:
Makes sense, I'll add a continue here